src/NUnitFramework/framework/Internal/Execution/TestWorker.cs

```diff
@@ -66,24 +65,29 @@ public class TestWorker
         /// <param name="apartmentState">The apartment state to use for running tests</param>
         public TestWorker(WorkItemQueue queue, string name, ApartmentState apartmentState)
         {
-            _readyQueue = queue;
-
-            _workerThread = new Thread(new ThreadStart(TestWorkerThreadProc));
-            _workerThread.Name = name;
-            _workerThread.SetApartmentState(apartmentState);
+            WorkQueue = queue;
+            Name = name;
+            Apartment = apartmentState;
```

Comment:
Can you run through the reasoning behind the changes to this file, for the ignorant? This is the only one I didn't follow. 😄